plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php

```diff
@@ -156,9 +168,13 @@ private function process_next_batch_for_single_processor( string $processor_clas
 		$still_pending   = count( $batch_processor->get_next_batch_to_process( 1 ) ) > 0;
 		if ( ( $error instanceof \Exception ) ) {
 			// The batch processing failed and no items were processed:
-			// reschedule the processing with a delay, and also throw the error
-			// so Action Scheduler will ignore the rescheduling if this happens repeatedly.
-			$this->schedule_batch_processing( $processor_class_name, true );
+			// reschedule the processing with a delay, unless this is a repeatead failure.
+			if ( $this->is_consistently_failing( $batch_processor ) ) {
+				$this->logger->error( "Batch processor {$batch_processor->get_name()} appears to be consistently failing. Execution won't be automatically re-attempted." );
```

Comment:
Could we add here (as additional data) the number of failed attempts and maybe also the date of the first and last attempts? Sure, this information can be inferred from the previous log entries; but it will be useful to have it condensed in one place, especially in case of recurring errors.

Also, maybe the source of the log could be set to something like `batch-processing`, to make it more easy to identify in the main logs page.

Reply:
Thanks for the input @Konamiman, @coreymckrill! I've made some adjustments to the logging to include more details and work better with the logger.

<img width="944" alt="Screenshot 2024-06-04 at 16 13 26" src="https://github.com/woocommerce/woocommerce/assets/184724/ecfe33f8-16a1-4134-8d9f-a939f6b3db43">
<img width="1171" alt="Screenshot 2024-06-04 at 16 16 04" src="https://github.com/woocommerce/woocommerce/assets/184724/d5779286-ed96-47a9-b2f7-9cc7ad9721d1">